integration-tests/src/test/java/org/apache/fineract/integrationtests/DelinquencyBucketsIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.ResponseSpecification;
+import java.util.ArrayList;
+import org.apache.fineract.client.models.GetDelinquencyBucketsResponse;
+import org.apache.fineract.client.models.GetDelinquencyRangesResponse;
+import org.apache.fineract.client.models.PostDelinquencyBucketResponse;
+import org.apache.fineract.client.models.PostDelinquencyRangeResponse;
+import org.apache.fineract.client.models.PutDelinquencyRangeResponse;
+import org.apache.fineract.integrationtests.common.Utils;
+import org.apache.fineract.integrationtests.common.products.DelinquencyBucketsHelper;
+import org.apache.fineract.integrationtests.common.products.DelinquencyRangesHelper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class DelinquencyBucketsIntegrationTest {
+
+    private ResponseSpecification responseSpec;
+    private RequestSpecification requestSpec;
+
+    @BeforeEach
+    public void setup() {
+        Utils.initializeRESTAssured();
+
+        requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+    }
+
+    @Test
+    public void testCreateDelinquencyRanges() {
+        // given
+        final String jsonRange = DelinquencyRangesHelper.getAsJSON(1, 3);
+
+        // when
+        final PostDelinquencyRangeResponse delinquencyRangeResponse01 = DelinquencyRangesHelper.createDelinquencyRange(requestSpec,
+                responseSpec, jsonRange);
+        final ArrayList<GetDelinquencyRangesResponse> ranges = DelinquencyRangesHelper.getDelinquencyRanges(requestSpec, responseSpec);
+
+        // then
+        assertNotNull(delinquencyRangeResponse01);
+        assertNotNull(ranges);
+        assertEquals(3, ranges.get(0).getMaximumAgeDays(), "Expected Max Age Days to 3");
+    }
+
+    @Test
+    public void testUpdateDelinquencyRanges() {
+        // given
+        String jsonRange = DelinquencyRangesHelper.getAsJSON(1, 3);
+        final PostDelinquencyRangeResponse delinquencyRangeResponse01 = DelinquencyRangesHelper.createDelinquencyRange(requestSpec,
+                responseSpec, jsonRange);
+        jsonRange = DelinquencyRangesHelper.getAsJSON(1, 7);
+
+        // when
+        final PutDelinquencyRangeResponse delinquencyRangeResponse02 = DelinquencyRangesHelper.updateDelinquencyRange(requestSpec,
+                responseSpec, delinquencyRangeResponse01.getResourceId(), jsonRange);
+        final GetDelinquencyRangesResponse range = DelinquencyRangesHelper.getDelinquencyRange(requestSpec, responseSpec,
+                delinquencyRangeResponse01.getResourceId());
+
+        // then
+        assertNotNull(delinquencyRangeResponse01);
```

Comment:
@josehernandezfintecheandomx Useless check. If the delinquencyRangeResponse01 is null, there will be a NPE at line 82

Reply:
Done!